hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/EventloopFactory.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.tpc;
+
+import com.hazelcast.internal.util.ThreadAffinity;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Supplier;
+
+import static com.hazelcast.internal.tpc.util.Preconditions.checkNotNegative;
+import static com.hazelcast.internal.tpc.util.Preconditions.checkNotNull;
+import static com.hazelcast.internal.tpc.util.Preconditions.checkPositive;
+import static java.lang.System.getProperty;
+
+/**
+ * A factory for {@link Eventloop} instances.
+ */
+public abstract class EventloopFactory {
+    public static final String NAME_LOCAL_QUEUE_CAPACITY = "hazelcast.tpc.localQueue.capacity";
```

Comment:
This Configuration is internal to the TPC engine. So none of this can be moved to Alto because that would create a coupling between the TPC engine and Alto. So what probably happens is that some of these properties will get copied to alto and the rest we modify using system properties.